src/python/WMCore/WMSpec/Steps/Fetchers/PileupFetcher.py

```diff
@@ -70,15 +74,21 @@ def _queryDbsAndGetPileupConfig(self, stepHelper, dbsReader, fakeSites):
             # each dataset input can generally be a list, iterate over dataset names
             blockDict = {}
```

Comment:
One thing I don't quite get is blockDict is outside of datasets loop. I am assuming there will be only one pileup dataset for each type. but if there are mulitple datasets in the same type, wouldn't this cause the problem by missing blocks for different dataset? But I leave as it is since that will require further down return data structure change.